core/domain/story_domain.py

```
@@ -1167,6 +1395,48 @@ def validate(self) -> None:
+                    raise utils.ValidationError(
+                        'Node %s is covered by multiple arcs.' % node_id
+                    )
+                covered_node_ids.add(node_id)
+
+        if len(arc_ids_list) > len(set(arc_ids_list)):
+            raise utils.ValidationError('Expected all arc ids to be distinct.')
+
+        if len(self.nodes) > 0 and len(self.arcs) == 0:
+            raise utils.ValidationError(
+                'Expected at least one arc when story has nodes.'
+            )
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)